meson.build

```
@@ -84,6 +84,8 @@ endif
 if host_machine.system() == 'darwin' and cc.has_link_argument('-Wl,-ld_classic')
   # New linker introduced in macOS 14 not working yet, see gh-19357 and gh-19387
   add_project_link_arguments('-Wl,-ld_classic', language : ['c', 'cpp', 'fortran'])
+  # Allow linker to strip unused symbols
+  add_project_link_arguments('-Wl,-dead_strip', language : ['c', 'cpp', 'fortran'])
 endif
```

Not sure this is necessary, but I note that the guard check here explicitly tests for availability of the existing linker argument before adding it. Perhaps we should do the same for `-Wl,-dead_strip`?
```suggestion
if host_machine.system() == 'darwin'
  if cc.has_link_argument('-Wl,-ld_classic')
    # New linker introduced in macOS 14 not working yet, see gh-19357 and gh-19387
    add_project_link_arguments('-Wl,-ld_classic', language : ['c', 'cpp', 'fortran'])
  endif
  if cc.has_link_argument('-Wl,-dead_strip')
    # Allow linker to strip unused symbols
    add_project_link_arguments('-Wl,-dead_strip', language : ['c', 'cpp', 'fortran'])
  endif
endif
```